github.com/cockroachdb/pebble/internal/metamorphic: TestMeta failed

github.com/cockroachdb/pebble/internal/metamorphic.TestMeta [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_Pebble_Metamorphic/12908272?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_Pebble_Metamorphic/12908272?buildTab=artifacts#meta) on refs/heads/master @ [4fc6ccd93ae2](https://github.com/cockroachdb/pebble/commits/4fc6ccd93ae2):


```
        snap29.Get("nuatsf@86") // [""] pebble: not found #3070
        iter28.SeekLT("xdlgtbn@23", "xdlgtbn@55") // [invalid] <nil> #3066
        db1.Set("fcylw@59", "jiqdhprwamipqzqozhw") // <nil> #3075
        db1.Set("fcylw@110", "fbqwhhlzcykseblqql") // <nil> #3085
        db1.Set("tajsdbi@77", "vtxyxjsjkw") // <nil> #3090
        db1.RangeKeyUnset("fcylw", "rleqb", "@111") // <nil> #3094
        iter37.Next("") // [true,"oqpa@51","udwndkhwwmuuqekyw",["oqpa@97","oqpa@36")=>{"@109"="xcnuzun","@105"="tiwjfzbjkdjcuqdlj"}] <nil> #3025
        iter37.SeekLT("xdlgtbn@45", "eehhcjtbdq@64") // [valid,"tajsdbi@108","haglhsuhtuxcbmu",["tajsdbi","tajsdbi@97")=>{""="qhwok","@105"="tiwjfzbjkdjcuqdlj","@104"="lmhxrf"}*] <nil> #3029
        iter37.SetBounds("kicfsyxild@71", "komvltqws@82") // <nil> #3033
        iter37.SeekLT("komvltqws@82", "") // [true,"komvltqws@89","jikzzmrfwo",["komvltqws","komvltqws@82")=>{"@109"="xcnuzun","@105"="tiwjfzbjkdjcuqdlj","@47"="segsq"}*] <nil> #3034
        iter35.SeekPrefixGE("oqpa@31") // [false] pebble: SeekPrefixGE supplied with key outside of upper bound #3130
        iter37.Next("") // [false] <nil> #3035
        iter37.First() // [true,"kicfsyxild@71","ndvremjnujyzmwyhqp",["kicfsyxild@71","komvltqws")=>{"@109"="xcnuzun","@47"="segsq"}*] <nil> #3039
        iter35.Prev("") // [false] pebble: SeekPrefixGE supplied with key outside of upper bound #3133
        iter37.SeekLT("tajsdbi@69", "") // [true,"komvltqws@89","jikzzmrfwo",["komvltqws","komvltqws@82")=>{"@109"="xcnuzun","@105"="tiwjfzbjkdjcuqdlj","@47"="segsq"}*] <nil> #3046
        iter37.Prev("qukudex@77") // [invalid] <nil> #3048
        iter35.Prev("xdlgtbn@1") // [invalid] pebble: SeekPrefixGE supplied with key outside of upper bound #3153
        iter37.Prev("kicfsyxild@45") // [valid,"komvltqws@90","vjpvhlf",["komvltqws","komvltqws@82")=>{"@109"="xcnuzun","@105"="tiwjfzbjkdjcuqdlj","@47"="segsq"}*] <nil> #3059
        iter37.SetBounds("oqpa@35", "qukudex@84") // <nil> #3060
        iter35.Prev("") // [false] pebble: SeekPrefixGE supplied with key outside of upper bound #3155
        iter35.SeekLT("komvltqws@6", "") // [true,"eehhcjtbdq@14","sxbiqkxhcw",["eehhcjtbdq","fcylw@110")=>{"@47"="segsq"}*] <nil> #3157
        iter37.SeekGE("oqpa@35", "") // [true,"oqpa@35",<no point>,["oqpa@35","owzuwjyif")=>{"@109"="xcnuzun","@105"="tiwjfzbjkdjcuqdlj","@101"="iw"}*] <nil> #3061
        iter35.SeekLT("xdlgtbn@4", "") // [true,"eehhcjtbdq@14","sxbiqkxhcw",["eehhcjtbdq","fcylw@110")=>{"@47"="segsq"}] <nil> #3174
        iter37.Prev("") // [false] <nil> #3062
        iter35.SeekGE("kicfsyxild@43", "") // [false] <nil> #3180
        iter35.Next("") // [false] <nil> #3187
        iter35.Prev("") // [true,"eehhcjtbdq@14","sxbiqkxhcw",["eehhcjtbdq","fcylw@110")=>{"@47"="segsq"}*] <nil> #3206
        iter28.Prev("") // [true,"xdlgtbn@62","nyohtqztoxds",["xdlgtbn","xdlgtbn@59")=>{"@103"="qhalfpswiibnaynmjhi"}*] <nil> #3078
        iter35.Prev("") // [true,"eehhcjtbdq@20","nmitcvqqf",["eehhcjtbdq","fcylw@110")=>{"@47"="segsq"}] <nil> #3214
        iter35.Next("") // [true,"eehhcjtbdq@14","sxbiqkxhcw",["eehhcjtbdq","fcylw@110")=>{"@47"="segsq"}] <nil> #3215
        iter28.SeekPrefixGE("zgtu@80") // [false] pebble: SeekPrefixGE supplied with key outside of upper bound #3081
        iter28.Last() // [true,"xdlgtbn@62","nyohtqztoxds",["xdlgtbn","xdlgtbn@59")=>{"@103"="qhalfpswiibnaynmjhi"}*] <nil> #3086
        snap28.Close() // <nil> #3095
        iter28.SeekLT("tajsdbi@50", "") // [true,"tajsdbi@76","wfehffiszp",["tajsdbi","uulpm")=>{"@104"="lmhxrf"}*] <nil> #3100
        iter28.SeekPrefixGE("eehhcjtbdq@79") // [true,"eehhcjtbdq@20","nmitcvqqf",<no range>*] <nil> #3102
        iter28.First() // [true,"eehhcjtbdq@20","nmitcvqqf",<no range>] <nil> #3112
        iter28.SeekPrefixGE("komvltqws@33") // [false] <nil> #3118
        iter28.SetBounds("zgtu@81", "zgtu@46") // <nil> #3121
        iter28.SeekGE("zgtu@81", "") // [true,"zgtu@81","kgwo",<no range>] <nil> #3122
        iter28.Prev("") // [false] <nil> #3123
        iter28.SetBounds("kicfsyxild@96", "komvltqws@33") // <nil> #3126
        iter28.SeekLT("komvltqws@33", "") // [true,"komvltqws@67","bmng",<no range>] <nil> #3127
        iter28.Next("") // [false] <nil> #3128
        iter28.Prev("") // [true,"komvltqws@67","bmng",<no range>] <nil> #3129
        iter28.SeekLT("kicfsyxild@45", "") // [true,"kicfsyxild@70","iqgggnxmzkhnogoykwr",<no range>] <nil> #3141
        iter28.Next("zgtu@76") // [valid,"kicfsyxild@45","hmtvaks",<no range>] <nil> #3148
        iter28.SeekGE("tajsdbi@57", "") // [false] <nil> #3151
        iter28.SetBounds("longzuolff@83", "xdlgtbn") // <nil> #3168
        iter28.SeekGE("oqpa@103", "qgdlmjs@104") // [valid,"oqpa@38","fuqatuikschfbdgvj",<no range>] <nil> #3172
        --- FAIL: TestMeta/execution/standard-009 (0.22s)
```
<details><summary>Help</summary>
<p>
To reproduce, try:

```bash
go test -tags 'invariants' -exec 'stress -p 1' -timeout 0 -test.v -run TestMeta$ ./internal/metamorphic -seed 1701322945034781021 -ops "uniform:5000-10000"
```
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestMeta.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

github.com/cockroachdb/pebble/internal/metamorphic.TestMeta [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_NightlyPebbleMetamorphicRace/12908267?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_NightlyPebbleMetamorphicRace/12908267?buildTab=artifacts#meta) on refs/heads/master @ [4fc6ccd93ae2](https://github.com/cockroachdb/pebble/commits/4fc6ccd93ae2):


```
        iter64.SeekLT("hzaiy@65", "") // [true,"cnugxffj",<no point>,["cnugxffj","fmrbukid")=>{""=""}*] <nil> #5316
        iter64.Prev("") // [false] <nil> #5317
        iter64.SeekLT("hzaiy@27", "") // [true,"cnugxffj",<no point>,["cnugxffj","fmrbukid")=>{""=""}*] <nil> #5320
        iter64.SeekPrefixGE("nuczhm@77") // [false] pebble: SeekPrefixGE supplied with key outside of upper bound #5322
        iter64.Prev("") // [false] pebble: SeekPrefixGE supplied with key outside of upper bound #5333
        iter64.Next("iarw@53") // [invalid] cannot use limit with prefix iteration #5344
        iter64.NextPrefix() // [false] cannot use limit with prefix iteration #5346
        iter64.InternalNext() // cannot use limit with prefix iteration #5366
        iter64.SeekPrefixGE("azkykfhi@35") // [false] pebble: SeekPrefixGE supplied with key outside of lower bound #5367
        iter64.SeekLT("sdrlpuqdqp@28", "") // [true,"cnugxffj",<no point>,["cnugxffj","fmrbukid")=>{""=""}*] <nil> #5368
        iter64.First() // [true,"cnugxffj",<no point>,["cnugxffj","fmrbukid")=>{""=""}] <nil> #5372
        iter64.InternalNext() // <nil> #5386
        iter64.Last() // [true,"cnugxffj",<no point>,["cnugxffj","fmrbukid")=>{""=""}] <nil> #5387
        iter64.Next("njhamz@92") // [invalid] <nil> #5389
        iter64.First() // [true,"cnugxffj",<no point>,["cnugxffj","fmrbukid")=>{""=""}*] <nil> #5392
        iter66.SeekGE("bcxvmpzduox@25", "") // [true,"bcxvmpzduox@25",<no point>,["bcxvmpzduox","lftwlnmb")=>{""="gottzabgbzbsceyy","@96"="fjbo"}*] <nil> #5381
        iter66.Next("") // [true,"bcxvmpzduox@17","spgghndv",["bcxvmpzduox","lftwlnmb")=>{""="gottzabgbzbsceyy","@96"="fjbo"}] <nil> #5397
        iter66.SeekLT("bcxvmpzduox@42", "bcxvmpzduox@43") // [valid,"bcxvmpzduox@45","xjjqol",["bcxvmpzduox","lftwlnmb")=>{""="gottzabgbzbsceyy","@96"="fjbo"}] <nil> #5398
        iter66.SetBounds("", "azkykfhi@13") // <nil> #5403
        iter66.SeekPrefixGE("awkohvz@29") // [true,"awkohvz@29","khrfawc",["awkohvz","awkohvz\x00")=>{"@40"="cebtif"}*] <nil> #5416
        iter66.SetBounds("hzaiy@18", "rasqgidaw@16") // <nil> #5421
        iter66.SeekGE("hzaiy@18", "") // [true,"hzaiy@18",<no point>,["hzaiy@18","lftwlnmb")=>{""="gottzabgbzbsceyy","@96"="fjbo"}*] <nil> #5422
        iter66.Next("") // [true,"iarw@90","clk",["hzaiy@18","lftwlnmb")=>{""="gottzabgbzbsceyy","@96"="fjbo"}] <nil> #5423
        iter66.SeekLT("bcxvmpzduox", "") // [false] <nil> #5432
        iter66.SeekPrefixGE("bcxvmpzduox@6") // [false] pebble: SeekPrefixGE supplied with key outside of lower bound #5435
        iter66.First() // [true,"hzaiy@18",<no point>,["hzaiy@18","lftwlnmb")=>{""="gottzabgbzbsceyy","@96"="fjbo"}*] <nil> #5441
        iter66.Last() // [true,"rasqgidaw@23","qea",["njhamz","rasqgidaw@16")=>{""="gottzabgbzbsceyy","@96"="fjbo"}*] <nil> #5445
        iter66.Next("") // [false] <nil> #5448
        iter66.SetBounds("", "lvdgpb@80") // <nil> #5461
        iter66.SeekLT("lvdgpb@80", "") // [true,"lftwlnmb@59","ednvil",["lftwlnmb","lvdgpb@80")=>{""="gottzabgbzbsceyy","@96"="fjbo","@73"="azlc"}*] <nil> #5462
        iter66.Next("") // [false] <nil> #5463
        iter66.Next("") // [false] <nil> #5464
        iter66.SeekLT("iarw@41", "") // [true,"iarw@61","",["bcxvmpzduox","lftwlnmb")=>{""="gottzabgbzbsceyy","@96"="fjbo"}*] <nil> #5467
        iter66.Next("") // [true,"iarw@40","jeicignbsvoijxbcp",["bcxvmpzduox","lftwlnmb")=>{""="gottzabgbzbsceyy","@96"="fjbo"}] <nil> #5468
        iter66.Prev("") // [true,"iarw@61","",["bcxvmpzduox","lftwlnmb")=>{""="gottzabgbzbsceyy","@96"="fjbo"}] <nil> #5477
        iter66.SeekGE("hzaiy@18", "vkruzryx@53") // [valid,"hzaiy@18",<no point>,["bcxvmpzduox","lftwlnmb")=>{""="gottzabgbzbsceyy","@96"="fjbo"}] <nil> #5483
        iter66.Prev("") // [true,"hzaiy@33","writxsfcu",["bcxvmpzduox","lftwlnmb")=>{""="gottzabgbzbsceyy","@96"="fjbo"}] <nil> #5484
        iter66.SetBounds("vkruzryx@44", "wmuinuqcko@55") // <nil> #5486
        iter66.SeekGE("vkruzryx@44", "") // [true,"vkruzryx@44",<no point>,["vkruzryx@44","wmuinuqcko")=>{"@96"="fjbo","@92"="uiuioahejvcr","@88"="uplaej","@83"="rrrkhfbfnhmogvzkuk","@79"="kkuywtv","@51"="lphdzzt","@45"="nxiqkijaennndcd"}*] <nil> #5487
        iter66.Prev("") // [false] <nil> #5488
        iter64.SeekLT("rasqgidaw@72", "") // [true,"cnugxffj",<no point>,["cnugxffj","fmrbukid")=>{""=""}] <nil> #5394
        iter64.SetBounds("hqrdfjpq", "ldsdzmhcn") // <nil> #5395
        iter64.SeekLT("ldsdzmhcn", "") // [true,"iihpfi@84","buzphtzruhsidk",["iihpfi","ldsdzmhcn")=>{""=""}*] <nil> #5396
        iter64.SetBounds("azkykfhi", "encbdianjvr") // <nil> #5399
        iter64.SeekGE("azkykfhi", "") // [true,"azkykfhi","wqxeqaz",["azkykfhi","azkykfhi@15")=>{""="nnpelijll","@73"="zvyfqz","@72"="bklxn","@68"="uvepr","@40"="cebtif"}*] <nil> #5400
        iter64.Prev("") // [false] <nil> #5401
        iter64.Next("") // [true,"azkykfhi","wqxeqaz",["azkykfhi","azkykfhi@15")=>{""="nnpelijll","@73"="zvyfqz","@72"="bklxn","@68"="uvepr","@40"="cebtif"}*] <nil> #5402
        iter64.Next("") // [true,"azkykfhi@81","xnnylytkwr",["azkykfhi","azkykfhi@15")=>{""="nnpelijll","@73"="zvyfqz","@72"="bklxn","@68"="uvepr","@40"="cebtif"}] <nil> #5408
        iter64.Next("") // [true,"azkykfhi@77","ionh",["azkykfhi","azkykfhi@15")=>{""="nnpelijll","@73"="zvyfqz","@72"="bklxn","@68"="uvepr","@40"="cebtif"}] <nil> #5434
        --- FAIL: TestMeta/execution/standard-025 (3.37s)
```
<details><summary>Help</summary>
<p>
To reproduce, try:

```bash
go test -tags 'invariants' -exec 'stress -p 1' -timeout 0 -test.v -run TestMeta$ ./internal/metamorphic -seed 1701323222806691300 -ops "uniform:5000-10000"
```
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestMeta.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

github.com/cockroachdb/pebble/internal/metamorphic.TestMeta [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_Pebble_Metamorphic/12925604?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_Pebble_Metamorphic/12925604?buildTab=artifacts#meta) on refs/heads/master @ [51fca96df08f](https://github.com/cockroachdb/pebble/commits/51fca96df08f):


```
        iter34.SeekPrefixGE("xbqyn@56") // [false] <nil> #2358
        iter37.SeekGE("yvgiwjowj@236", "") // [false] <nil> #2476
        iter34.SeekLT("ybmwniowp@260", "gqluck@173") // [valid,"xbqyn@71","obfsbfs",["xbqyn@165","xbqyn@64")=>{"@282"="u","@275"="dsqwmfsztrxpenjuwhd","@173"="alghaulqvimaju"}*] <nil> #2360
        iter37.First() // [true,"snhbdledmk@279",<no point>,["snhbdledmk@279","xhjld")=>{""="dvoukrxqamlr"}*] <nil> #2478
        iter35.Next("") // [false] <nil> #2470
        // INFO: [JOB 168] sstable deleted 000127
        iter34.SeekPrefixGE("xbqyn@191") // [true,"xbqyn@165",<no point>,["xbqyn@165","xbqyn@64")=>{"@282"="u","@275"="dsqwmfsztrxpenjuwhd","@173"="alghaulqvimaju"}] <nil> #2361
        db1.Set("ybmwniowp@285", "rjcbjxdmczlnup") // <nil> #2393
        db1.Set("ihtqyywy@247", "fed") // <nil> #2400
        iter34.Prev("") // [false] pebble: unsupported reverse prefix iteration #2363
        db1.RangeKeyUnset("ihtqyywy", "jdjwxjbjpuu", "@278") // <nil> #2402
        db1.Set("xbqyn@33", "rgamaux") // <nil> #2405
        iter34.Prev("") // [false] pebble: unsupported reverse prefix iteration #2364
        iter34.SetBounds("xbqyn@1", "yvgiwjowj@260") // <nil> #2365
        db1.Get("xbqyn@38") // [""] pebble: not found #2409
        db1.Set("ererx@267", "xcimywlvjtgcaiiv") // <nil> #2415
        iter34.SeekGE("xbqyn@1", "") // [true,"xbqyn@1",<no point>,["xbqyn@1","xhjld")=>{"@282"="u","@275"="dsqwmfsztrxpenjuwhd","@173"="alghaulqvimaju"}*] <nil> #2366
        snap29.Get("pquhrevor@275") // ["sfymtstjeiqvu"] <nil> #2419
        iter34.Prev("") // [false] <nil> #2367
        iter34.Prev("") // [false] <nil> #2370
        db1.Get("xbqyn@34") // [""] pebble: not found #2422
        db1.Set("xhjld@42", "xwthtkgv") // <nil> #2423
        iter34.Last() // [true,"ybmwniowp",<no point>,["ybmwniowp","yvgiwjowj")=>{"@271"="pvxnkgbkgbfbr","@263"="derschtcgfxwkzjf","@258"="a","@114"="vhwogtm"}*] <nil> #2373
        db1.Get("yvgiwjowj@232") // [""] pebble: not found #2425
        snap30 = db1.NewSnapshot("gqluck", "jdjwxjbjpuu", "poabgv", "pquhrevor", "pwhsqessb", "xbqyn", "ybmwniowp", "yvgiwjowj") #2428
        db1.Set("ererx@187", "vcgh") // <nil> #2430
        iter34.SeekGE("xbqyn@195", "") // [true,"xbqyn@1",<no point>,["xbqyn@1","xhjld")=>{"@282"="u","@275"="dsqwmfsztrxpenjuwhd","@173"="alghaulqvimaju"}*] <nil> #2375
        iter34.NextPrefix() // [false] NextPrefix not permitted with upper bound yvgiwjowj@260 #2377
        iter37.SeekPrefixGE("xbqyn@42") // [true,"xbqyn@42","oqhxcbiytivzijya",["xbqyn","xbqyn\x00")=>{""="dvoukrxqamlr"}*] <nil> #2484
        iter37.SetBounds("xhjld@281", "yvgiwjowj@253") // <nil> #2486
        iter34.SeekPrefixGE("xbqyn@247") // [true,"xbqyn@1",<no point>,["xbqyn@1","xbqyn\x00")=>{"@282"="u","@275"="dsqwmfsztrxpenjuwhd","@173"="alghaulqvimaju"}*] <nil> #2378
        iter34.SetBounds("", "") // <nil> #2381
        iter37.SeekLT("pwhsqessb@274", "") // [false] <nil> #2487
        iter34.SeekLT("yvgiwjowj@256", "") // [true,"ybmwniowp",<no point>,["ybmwniowp","yvgiwjowj")=>{"@271"="pvxnkgbkgbfbr","@263"="derschtcgfxwkzjf","@258"="a","@114"="vhwogtm"}*] <nil> #2382
        iter34.First() // [true,"ererx",<no point>,["ererx","exolmtmh")=>{"@283"="msjapfvo"}*] <nil> #2383
        iter37.SetBounds("gqluck@244", "xbqyn@33") // <nil> #2489
        iter34.SeekPrefixGE("poabgv@194") // [true,"poabgv@194",<no point>,["poabgv","poabgv\x00")=>{"@202"="s"}*] <nil> #2385
        iter37.SeekLT("xbqyn@33", "") // [true,"xbqyn@42","oqhxcbiytivzijya",["rrmuxabyi","xbqyn@33")=>{""="dvoukrxqamlr"}*] <nil> #2490
        iter34.SeekPrefixGE("pwhsqessb@274") // [true,"pwhsqessb@274",<no point>,["pwhsqessb","pwhsqessb\x00")=>{""="r","@202"="s"}*] <nil> #2386
        iter37.Next("") // [false] <nil> #2491
        iter37.Next("") // [false] <nil> #2492
        iter34.Prev("") // [false] pebble: unsupported reverse prefix iteration #2388
        iter37.Prev("") // [true,"xbqyn@42","oqhxcbiytivzijya",["rrmuxabyi","xbqyn@33")=>{""="dvoukrxqamlr"}*] <nil> #2493
        iter34.Last() // [true,"yvgiwjowj@61","lehp",<no range>] <nil> #2389
        iter37.First() // [true,"gqluck@244","lhmsvehvkupjek",["gqluck@244","iezbsede")=>{""="","@283"="msjapfvo","@202"="s"}*] <nil> #2501
        iter34.SeekLT("xbqyn@165", "") // [true,"xbqyn@200","moscsvonjjfktwdv",<no range>] <nil> #2394
        iter37.SeekGE("pquhrevor@263", "") // [true,"rrmuxabyi",<no point>,["rrmuxabyi","xbqyn@33")=>{""="dvoukrxqamlr"}*] <nil> #2506
        iter37.Next("") // [true,"rrmuxabyi@284","nptzrqrunmlsh",["rrmuxabyi","xbqyn@33")=>{""="dvoukrxqamlr"}] <nil> #2509
        iter37.Next("") // [true,"rrmuxabyi@97","nlfbllzlzmhyq",["rrmuxabyi","xbqyn@33")=>{""="dvoukrxqamlr"}] <nil> #2510
        --- FAIL: TestMeta/execution/standard-008 (0.21s)
```
<details><summary>Help</summary>
<p>
To reproduce, try:

```bash
go test -tags 'invariants' -exec 'stress -p 1' -timeout 0 -test.v -run TestMeta$ ./internal/metamorphic -seed 1701407438584764260 -ops "uniform:5000-10000"
```
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestMeta.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

github.com/cockroachdb/pebble/internal/metamorphic.TestMeta [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_NightlyPebbleMetamorphicRace/12925605?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_NightlyPebbleMetamorphicRace/12925605?buildTab=artifacts#meta) on refs/heads/master @ [51fca96df08f](https://github.com/cockroachdb/pebble/commits/51fca96df08f):


```
        iter50.SeekPrefixGE("dmfg@13") // [true,"dmfg@13","klsg",["dmfg","dmfg\x00")=>{"@157"="wson","@149"="kiipsisubucpprvpkxu"}*] <nil> #4455
        iter50.Last() // [true,"xutwvknn@16","yuvdbt",<no range>*] <nil> #4461
        iter50.SeekGE("mqrnztk@136", "") // [true,"mqrnztk@136","otvftejjnnunwlwyi",["mqrnztk","mqrnztk@88")=>{"@157"="wson","@152"="qvklzjhreqwkgm","@150"="ofluqyrp","@149"="kiipsisubucpprvpkxu","@144"="pexxvy","@122"="rdaaoiixkwqxhkqn","@103"="hzlajco","@33"="cgwbmgfmlnbuuxym"}*] <nil> #4462
        iter50.SeekPrefixGE("fcba@108") // [true,"fcba@108","ljgpwjkqnpnlt",["fcba","fcba\x00")=>{"@149"="kiipsisubucpprvpkxu"}*] <nil> #4472
        iter50.Last() // [true,"xutwvknn@16","yuvdbt",<no range>*] <nil> #4479
        iter50.First() // [true,"ctcmxwqzk",<no point>,["ctcmxwqzk","czxbmlewh")=>{"@150"="ksd","@137"="yqsmqowrcvf","@125"="seotm","@70"="uixvekylrj"}*] <nil> #4493
        iter48.Prev("") // [false] <nil> #4027
        iter48.SetBounds("mqrnztk@100", "wwdc@36") // <nil> #4032
        iter48.SetBounds("fcba@119", "iumcimiom@20") // <nil> #4034
        iter48.SeekGE("mqrnztk@133", "") // [false] <nil> #4035
        iter48.SeekPrefixGE("xutwvknn@16") // [false] pebble: SeekPrefixGE supplied with key outside of upper bound #4036
        iter48.SeekGE("ctcmxwqzk@154", "") // [true,"fcba@119",<no point>,["fcba@119","iumcimiom")=>{"@149"="kiipsisubucpprvpkxu"}*] <nil> #4037
        iter48.Next("") // [true,"fcba@113","",["fcba@119","iumcimiom")=>{"@149"="kiipsisubucpprvpkxu"}] <nil> #4041
        iter48.SetBounds("fcba@119", "fcba@119") // <nil> #4046
        iter48.SeekLT("mqrnztk@73", "") // [false] <nil> #4047
        iter48.SeekGE("hwioffi@105", "") // [false] <nil> #4049
        iter48.Last() // [false] <nil> #4050
        iter48.First() // [false] <nil> #4055
        iter48.NextPrefix() // [false] NextPrefix not permitted with upper bound fcba@119 #4057
        iter50.Prev("") // [false] <nil> #4496
        iter50.Next("") // [true,"ctcmxwqzk",<no point>,["ctcmxwqzk","czxbmlewh")=>{"@150"="ksd","@137"="yqsmqowrcvf","@125"="seotm","@70"="uixvekylrj"}*] <nil> #4499
        iter50.SeekLT("iumcimiom@117", "ctcmxwqzk@43") // [valid,"iumcimiom@130","umpki",["iumcimiom","mqrnztk")=>{"@157"="wson","@149"="kiipsisubucpprvpkxu"}*] <nil> #4507
        iter50.SeekGE("rnvllznt@10", "") // [true,"rnvllznt@10",<no point>,["qnenxccpc","wwdc")=>{"@149"="kiipsisubucpprvpkxu"}*] <nil> #4510
        iter50.SeekGE("rnvllznt@88", "") // [true,"rnvllznt@88",<no point>,["qnenxccpc","wwdc")=>{"@149"="kiipsisubucpprvpkxu"}] <nil> #4520
        iter50.SeekGE("iumcimiom@40", "wwdc@103") // [valid,"iumcimiom@40","zg",["iumcimiom","mqrnztk")=>{"@157"="wson","@149"="kiipsisubucpprvpkxu"}*] <nil> #4536
        iter50.SeekLT("rnvllznt@11", "") // [true,"rnvllznt@137","h",["qnenxccpc","wwdc")=>{"@149"="kiipsisubucpprvpkxu"}*] <nil> #4545
        iter50.SeekLT("mqrnztk@39", "ctcmxwqzk@59") // [valid,"mqrnztk@88",<no point>,["mqrnztk@88","oiogcgrnsx")=>{"@157"="wson","@149"="kiipsisubucpprvpkxu","@103"="hzlajco"}*] <nil> #4552
        iter50.Next("iumcimiom@27") // [invalid] <nil> #4553
        iter50.Last() // [true,"xutwvknn@16","yuvdbt",<no range>] <nil> #4558
        iter50.Next("") // [false] <nil> #4559
        iter48.Next("qnenxccpc@142") // [invalid] NextPrefix not permitted with upper bound fcba@119 #4058
        iter48.Prev("") // [false] NextPrefix not permitted with upper bound fcba@119 #4060
        iter48.First() // [false] <nil> #4063
        iter48.SeekGE("wwdc@124", "wwdc@67") // [invalid] <nil> #4064
        iter48.SeekLT("ctcmxwqzk@90", "") // [false] <nil> #4070
        iter48.Last() // [false] <nil> #4071
        iter48.SeekPrefixGE("norjyubew@50") // [false] pebble: SeekPrefixGE supplied with key outside of upper bound #4074
        iter48.SeekGE("hwioffi@140", "") // [false] <nil> #4075
        iter48.SeekGE("dmfg@13", "") // [false] <nil> #4077
        iter48.SetBounds("", "ctcmxwqzk@97") // <nil> #4078
        iter48.SeekLT("ctcmxwqzk@97", "") // [true,"ctcmxwqzk@99","mrctzhiqqln",["ctcmxwqzk","ctcmxwqzk@97")=>{"@150"="ksd","@137"="yqsmqowrcvf","@125"="seotm","@70"="uixvekylrj"}*] <nil> #4079
        iter48.Next("") // [false] <nil> #4080
        iter48.Prev("") // [true,"ctcmxwqzk@99","mrctzhiqqln",["ctcmxwqzk","ctcmxwqzk@97")=>{"@150"="ksd","@137"="yqsmqowrcvf","@125"="seotm","@70"="uixvekylrj"}*] <nil> #4081
        iter48.SeekGE("xutwvknn@16", "") // [false] <nil> #4082
        iter48.SetBounds("hwioffi@106", "rnvllznt@120") // <nil> #4086
        iter50.SeekGE("mqrnztk@96", "norjyubew@90") // [valid,"mqrnztk@96",<no point>,["mqrnztk","mqrnztk@88")=>{"@157"="wson","@152"="qvklzjhreqwkgm","@150"="ofluqyrp","@149"="kiipsisubucpprvpkxu","@144"="pexxvy","@122"="rdaaoiixkwqxhkqn","@103"="hzlajco","@33"="cgwbmgfmlnbuuxym"}*] <nil> #4567
        iter48.SeekPrefixGE("mqrnztk@93") // [true,"mqrnztk@93",<no point>,["mqrnztk","mqrnztk@88")=>{"@157"="wson","@152"="qvklzjhreqwkgm","@150"="ofluqyrp","@149"="kiipsisubucpprvpkxu","@144"="pexxvy","@122"="rdaaoiixkwqxhkqn","@103"="hzlajco","@33"="cgwbmgfmlnbuuxym"}*] <nil> #4087
        iter48.SetBounds("hwioffi@106", "hwioffi@106") // <nil> #4091
        iter50.Prev("") // [true,"mqrnztk@98","yjjenqyjny",["mqrnztk","mqrnztk@88")=>{"@157"="wson","@152"="qvklzjhreqwkgm","@150"="ofluqyrp","@149"="kiipsisubucpprvpkxu","@144"="pexxvy","@122"="rdaaoiixkwqxhkqn","@103"="hzlajco","@33"="cgwbmgfmlnbuuxym"}] <nil> #4585
        --- FAIL: TestMeta/execution/standard-016 (1.97s)
```
<details><summary>Help</summary>
<p>
To reproduce, try:

```bash
go test -tags 'invariants' -exec 'stress -p 1' -timeout 0 -test.v -run TestMeta$ ./internal/metamorphic -seed 1701412087825623789 -ops "uniform:5000-10000"
```
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestMeta.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

github.com/cockroachdb/pebble/internal/metamorphic.TestMeta [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_Pebble_Metamorphic/12942546?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_Pebble_Metamorphic/12942546?buildTab=artifacts#meta) on refs/heads/master @ [51fca96df08f](https://github.com/cockroachdb/pebble/commits/51fca96df08f):


```
        iter59.InternalNext() // pebble: CanDeterministicallySingleDelete called twice #5172
        iter59.Prev("") // [false] switching from reverse to forward via internalNext is prohibited #5174
        iter59.Close() // switching from reverse to forward via internalNext is prohibited #5176
        snap49.Close() // <nil> #5177
        db1.Set("yjtlhfyhzum@137", "hxddhp") // <nil> #5178
        // INFO: [JOB 118] sstable deleted 000073
        // INFO: [JOB 118] sstable deleted 000074
        // INFO: [JOB 120] sstable deleted 000085
        // INFO: [JOB 120] sstable deleted 000086
        // INFO: [JOB 120] sstable deleted 000089
        snap48.Get("uphuybsdeh@51") // [""] pebble: not found #5179
        db1.SingleDelete("xjbkie@141", false /* maybeReplaceDelete */) // <nil> #5180
        db1.Get("mknhzwuh@135") // ["ypmliqimfw"] <nil> #5181
        db1.Set("fhtbwpszlpg@41", "eauyjgvgaacnowsmzxy") // <nil> #5182
        db1.SingleDelete("lwok@9", false /* maybeReplaceDelete */) // <nil> #5183
        snap48.Get("ebljo@55") // [""] pebble: not found #5184
        db1.RangeKeySet("fhtbwpszlpg", "szqq", "@115", "bltnajwmr") // <nil> #5185
        db1.Get("zpnclwms@94") // ["rzputc"] <nil> #5186
        snap48.Get("ebljo@110") // ["iwkzsk"] <nil> #5187
        db1.Get("ebljo@55") // [""] pebble: not found #5188
        snap48.Get("tvtw@64") // [""] pebble: not found #5189
        snap48.Close() // <nil> #5190
        db1.RangeKeyUnset("obdyvothf", "tvtw", "@109") // <nil> #5191
        db1.Get("tvtw@105") // [""] pebble: not found #5192
        db1.Set("uwpuyhaeb@27", "jkzl") // <nil> #5193
        // INFO: [JOB 121] WAL created 000093
        // INFO: [JOB 122] flushing 1 memtable (2.9KB) to L0
        // INFO: [JOB 122] flushing: sstable created 000094
        // INFO: [JOB 122] flushed 1 memtable (2.9KB) to L0 [000094] (3.9KB), in 0.0s (0.0s total), output rate 11MB/s
        // INFO: [JOB 123] compacting(default) L0 [000094] (3.9KB) Score=0.00 + L6 [000092] (4.2KB) Score=0.00; OverlappingRatio: Single 1.08, Multi 0.00
        // INFO: [JOB 123] compacting: sstable created 000095
        // INFO: [JOB 123] compacted(default) L0 [000094] (3.9KB) Score=0.00 + L6 [000092] (4.2KB) Score=0.00 -> L6 [000095] (5.6KB), in 0.0s (0.0s total), output rate 10MB/s
        // INFO: [JOB 122] WAL deleted 000088
        db1.Compact("ipknpbopm@130", "uwpuyhaeb@59", false /* parallelize */) // <nil> #5194
        iter60 = db1.NewIter("", "", 2 /* key types */, 0, 0, false /* use L6 filters */, "@62" /* masking suffix */) // <nil> #5195
        // INFO: [JOB 123] sstable deleted 000092
        // INFO: [JOB 123] sstable deleted 000094
        iter60.Next("") // [false] <nil> #5196
        db1.Get("ulrvsjmilz@75") // [""] pebble: not found #5198
        db1.Get("ebljo@86") // ["szuvoaaeamehd"] <nil> #5206
        db1.Set("kyaalryvb@129", "iqq") // <nil> #5211
        iter60.SeekLT("obdyvothf@94", "") // [true,"obdyvothf@105","n",["nywcwoha","scjb")=>{"@115"="bltnajwmr"}*] <nil> #5197
        iter60.Prev("") // [true,"obdyvothf@140","amnppyp",["nywcwoha","scjb")=>{"@115"="bltnajwmr"}] <nil> #5199
        iter60.SeekLT("eepnozt@132", "") // [true,"eepnozt@138","bmfnd",["eepnozt","eepnozt@121")=>{"@136"="hjurzdfxn","@135"="fn","@122"="opqgeypycngvdjzhx","@118"="ftvjhtwpviar","@102"="","@99"="k","@79"="jprpcybxkenmtai","@77"="wgrwvvacqb","@75"="qobpehyfy","@67"="zyesbvsozo"}*] <nil> #5200
        db1.Set("ulrvsjmilz@15", "eomgckrkmhhsttizlsl") // <nil> #5216
        db1.Get("ebljo@77") // [""] pebble: not found #5219
        db1.Set("uphuybsdeh@90", "omcogysutppbihqleaf") // <nil> #5227
        db1.Get("ipknpbopm@12") // [""] pebble: not found #5229
        db1.Set("obdyvothf@109", "rclzscuelxvwtsq") // <nil> #5243
        --- FAIL: TestMeta/execution/standard-027 (0.42s)
```
<details><summary>Help</summary>
<p>
To reproduce, try:

```bash
go test -tags 'invariants' -exec 'stress -p 1' -timeout 0 -test.v -run TestMeta$ ./internal/metamorphic -seed 1701493721435321018 -ops "uniform:5000-10000"
```
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestMeta.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

github.com/cockroachdb/pebble/internal/metamorphic.TestMeta [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_NightlyPebbleMetamorphicRace/12942547?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_NightlyPebbleMetamorphicRace/12942547?buildTab=artifacts#meta) on refs/heads/master @ [51fca96df08f](https://github.com/cockroachdb/pebble/commits/51fca96df08f):


```
        iter47.SetBounds("rirx@119", "sickplvypm@157") // <nil> #3032
        iter46.First() // [false] <nil> #3009
        iter44.Next("bmydsgx@25") // [invalid] cannot use limit with prefix iteration #3015
        iter46.SetBounds("gwxpuptrc", "hlqvpunptvp") // <nil> #3011
        iter44.Next("") // [false] cannot use limit with prefix iteration #3020
        iter44.Prev("") // [false] cannot use limit with prefix iteration #3036
        iter46.SeekGE("gwxpuptrc", "") // [true,"gwxpuptrc",<no point>,["gwxpuptrc","hlqvpunptvp")=>{"@173"="wwhspzokunrbsjkk","@108"="zfjz"}*] <nil> #3012
        iter46.Prev("") // [false] <nil> #3013
        iter47.SeekGE("rirx@119", "") // [false] <nil> #3033
        iter46.Next("") // [true,"gwxpuptrc",<no point>,["gwxpuptrc","hlqvpunptvp")=>{"@173"="wwhspzokunrbsjkk","@108"="zfjz"}*] <nil> #3014
        iter47.Prev("") // [false] <nil> #3034
        iter47.Next("") // [false] <nil> #3035
        iter47.SeekLT("ogphvi@36", "") // [false] <nil> #3037
        iter45.SeekLT("upszpk@161", "") // [true,"ogphvi@95",<no point>,["ogphvi@95","rirx")=>{"@108"="zfjz"}*] <nil> #3004
        iter47.First() // [false] <nil> #3039
        iter47.SeekPrefixGE("snqbjudt@104") // [false] pebble: SeekPrefixGE supplied with key outside of upper bound #3040
        iter47.SeekLT("rirx@65", "") // [false] <nil> #3048
        iter47.Prev("evchsvolfc@140") // [invalid] <nil> #3057
        iter47.SeekGE("lbtivmseqp@88", "") // [false] <nil> #3059
        iter47.First() // [false] <nil> #3060
        iter47.SeekGE("snqbjudt@117", "zrfeelqds@55") // [invalid] <nil> #3064
        iter47.Last() // [false] <nil> #3069
        iter45.Next("") // [false] <nil> #3005
        snap35.Get("bmydsgx@85") // [""] pebble: not found #3038
        iter46.SetBounds("ghoglmwn", "idkozbhu") // <nil> #3044
        iter47.First() // [false] <nil> #3074
        iter47.First() // [false] <nil> #3075
        iter47.SetBounds("", "bmydsgx@27") // <nil> #3080
        db1.Get("upszpk@161") // [""] pebble: not found #3041
        iter44.Prev("") // [false] cannot use limit with prefix iteration #3042
        db1.Set("evchsvolfc@166", "aisedxdikexmrzgethr") // <nil> #3043
        iter44.Prev("") // [false] cannot use limit with prefix iteration #3050
        iter47.SeekGE("zrfeelqds@150", "") // [false] <nil> #3087
        iter47.SeekGE("inscmbf@112", "") // [false] <nil> #3090
        iter46.SeekGE("ghoglmwn", "") // [true,"ghoglmwn",<no point>,["ghoglmwn","gpyelfxvzt")=>{""="qjnzyptw","@170"="prgr","@158"="slg","@142"="gcaksjmpnkotlk","@129"="updauaqjqiptx","@108"="zfjz"}*] <nil> #3045
        iter47.SeekPrefixGE("bmydsgx@57") // [true,"bmydsgx@34",<no point>,["bmydsgx@34","bmydsgx@27")=>{""="qjnzyptw","@170"="prgr","@158"="slg","@141"="knysluflkiay","@131"="rvogvwqeeuhsalrh","@129"="updauaqjqiptx","@128"="uuiegowggjqa"}*] <nil> #3092
        iter47.SeekGE("bmydsgx@20", "") // [false] <nil> #3094
        iter46.Prev("") // [false] <nil> #3046
        iter47.Next("") // [false] <nil> #3098
        iter47.Prev("") // [true,"bmydsgx@34",<no point>,["bmydsgx@34","bmydsgx@27")=>{""="qjnzyptw","@170"="prgr","@158"="slg","@141"="knysluflkiay","@131"="rvogvwqeeuhsalrh","@129"="updauaqjqiptx","@128"="uuiegowggjqa"}*] <nil> #3103
        iter47.SeekPrefixGE("snqbjudt@141") // [false] pebble: SeekPrefixGE supplied with key outside of upper bound #3104
        iter47.SetBounds("ogphvi@39", "rirx@106") // <nil> #3111
        iter45.Prev("") // [true,"ogphvi@95",<no point>,["ogphvi@95","rirx")=>{"@108"="zfjz"}*] <nil> #3006
        iter46.Next("") // [true,"ghoglmwn",<no point>,["ghoglmwn","gpyelfxvzt")=>{""="qjnzyptw","@170"="prgr","@158"="slg","@142"="gcaksjmpnkotlk","@129"="updauaqjqiptx","@108"="zfjz"}*] <nil> #3047
        iter46.SeekLT("sickplvypm@125", "") // [true,"hoppa@119","ukilghkqjdsblauewfm",<no range>*] <nil> #3049
        iter46.Close() // <nil> #3051
        iter45.Next("") // [false] <nil> #3018
        iter44.SeekGE("ogphvi@16", "") // [true,"ogphvi@16",<no point>,["hoppa","rirx")=>{"@108"="zfjz"}*] <nil> #3053
        iter44.First() // [true,"bmydsgx","fgvssmogdmr",["bmydsgx","bmydsgx@116")=>{""="qjnzyptw","@170"="prgr","@158"="slg","@141"="knysluflkiay","@131"="rvogvwqeeuhsalrh","@129"="updauaqjqiptx","@128"="uuiegowggjqa"}*] <nil> #3054
        --- FAIL: TestMeta/execution/random-015 (5.51s)
```
<details><summary>Help</summary>
<p>
To reproduce, try:

```bash
go test -tags 'invariants' -exec 'stress -p 1' -timeout 0 -test.v -run TestMeta$ ./internal/metamorphic -seed 1701503380410848831 -ops "uniform:5000-10000"
```
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestMeta.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

github.com/cockroachdb/pebble/internal/metamorphic.TestMeta [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_Pebble_Metamorphic/12952560?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_Pebble_Metamorphic/12952560?buildTab=artifacts#meta) on refs/heads/master @ [51fca96df08f](https://github.com/cockroachdb/pebble/commits/51fca96df08f):


```
        db1.Set("qaqqkb@43", "sphw") // <nil> #1473
        iter5.Next("ennkscidui@12") // [invalid] cannot use limit with prefix iteration #1495
        iter18.SeekPrefixGE("yakwsgcg@9") // [false] pebble: SeekPrefixGE supplied with key outside of upper bound #1573
        iter13.SetBounds("qsetvzbbvkh", "raygiunieui") // <nil> #1552
        iter5.Prev("") // [false] cannot use limit with prefix iteration #1497
        iter5.InternalNext() // cannot use limit with prefix iteration #1507
        iter5.SetBounds("ennkscidui", "ennkscidui@36") // <nil> #1522
        snap8.Get("cjovt@39") // [""] pebble: not found #1476
        iter18.SeekGE("qaqqkb@2", "") // [true,"qaqqkb@2",<no point>,["qaqqkb@6","rehgwfytwkt")=>{"@31"="o","@23"="y"}*] <nil> #1586
        iter5.SeekLT("ennkscidui@36", "") // [true,"ennkscidui",<no point>,["ennkscidui","ennkscidui@36")=>{"@33"="tqoyvutdnh","@31"="o","@28"="","@23"="y"}*] <nil> #1523
        iter18.Prev("") // [true,"qaqqkb@5","dlinrfrsmpwarmxwih",["qaqqkb@6","rehgwfytwkt")=>{"@31"="o","@23"="y"}] <nil> #1597
        iter5.Next("") // [false] <nil> #1524
        iter19 = iter18.Clone(false, "qaqqkb", "ximenszz", 2 /* key types */, 0, 0, false /* use L6 filters */, "@17" /* masking suffix */) // <nil> #1615
        snap3.Get("qaqqkb@23") // [""] pebble: not found #1478
        iter13.SeekLT("raygiunieui", "") // [true,"qsetvzbbvkh",<no point>,["qsetvzbbvkh","raygiunieui")=>{"@31"="o","@23"="y"}*] <nil> #1553
        iter13.Next("") // [false] <nil> #1554
        iter5.Prev("") // [true,"ennkscidui",<no point>,["ennkscidui","ennkscidui@36")=>{"@33"="tqoyvutdnh","@31"="o","@28"="","@23"="y"}*] <nil> #1525
        iter13.Next("") // [false] <nil> #1555
        iter5.SeekPrefixGE("qaqqkb@26") // [false] pebble: SeekPrefixGE supplied with key outside of upper bound #1531
        iter18.Last() // [true,"tbrm@27","kskrdjghregahqptg",<no range>*] <nil> #1617
        iter13.Prev("") // [true,"qsetvzbbvkh",<no point>,["qsetvzbbvkh","raygiunieui")=>{"@31"="o","@23"="y"}*] <nil> #1556
        iter18.Next("") // [false] <nil> #1620
        iter13.Next("ennkscidui@11") // [invalid] <nil> #1557
        iter18.SetBounds("tbkonpjhz", "yakwsgcg") // <nil> #1648
        db1.Set("qifxgtviq@43", "advptcpcchxhyofcsjp") // <nil> #1482
        iter13.SeekLT("rehgwfytwkt@12", "") // [true,"qsetvzbbvkh",<no point>,["qsetvzbbvkh","raygiunieui")=>{"@31"="o","@23"="y"}*] <nil> #1558
        iter13.SetBounds("idcjfqkztg", "iuvrhjvhmy") // <nil> #1566
        iter19.SeekGE("rehgwfytwkt@22", "") // [true,"rehgwfytwkt@22",<no point>,["rehgwfytwkt","tbrm")=>{"@31"="o","@23"="y","@10"="j"}*] <nil> #1616
        iter18.SeekGE("tbkonpjhz", "") // [true,"tbkonpjhz",<no point>,["tbkonpjhz","tbrm")=>{"@31"="o","@23"="y","@10"="j"}*] <nil> #1649
        iter19.Prev("") // [true,"rehgwfytwkt@31","xepfnkwgipckdftpub",["rehgwfytwkt","tbrm")=>{"@31"="o","@23"="y","@10"="j"}] <nil> #1618
        snap11.Get("zxfrhisycjj@12") // [""] pebble: not found #1488
        iter19.Next("") // [true,"rehgwfytwkt@12","ml",["rehgwfytwkt","tbrm")=>{"@31"="o","@23"="y","@10"="j"}] <nil> #1619
        db1.Set("jtgfxfps@7", "s") // <nil> #1491
        db1.Set("ennkscidui@28", "mjryqqzvybzwa") // <nil> #1492
        iter18.Prev("") // [false] <nil> #1650
        iter18.Prev("") // [false] <nil> #1651
        db1.Get("ennkscidui@11") // [""] pebble: not found #1498
        iter19.SeekLT("qaqqkb@11", "qaqqkb@27") // [valid,"qaqqkb@19","no",["qaqqkb","qaqqkb@6")=>{"@31"="o","@23"="y","@1"="karghr"}*] <nil> #1624
        snap10.Get("qaqqkb@7") // ["jlyyndjpteoxoq"] <nil> #1499
        iter19.Last() // [true,"tbrm@27","kskrdjghregahqptg",<no range>*] <nil> #1638
        iter13.SeekLT("iuvrhjvhmy", "") // [true,"idcjfqkztg",<no point>,["idcjfqkztg","iuvrhjvhmy")=>{"@33"="tqoyvutdnh","@31"="o","@23"="y"}*] <nil> #1567
        iter13.Prev("") // [false] <nil> #1568
        db1.SingleDelete("ennkscidui@21", false /* maybeReplaceDelete */) // <nil> #1500
        iter19.SeekGE("cjovt@30", "") // [true,"qaqqkb","agrhuoou",["qaqqkb","qaqqkb@6")=>{"@31"="o","@23"="y","@1"="karghr"}*] <nil> #1644
        db1.Set("ennkscidui@6", "mm") // <nil> #1505
        db1.Set("hvssubrba@22", "") // <nil> #1506
        db1.RangeKeySet("hvssubrba", "qaqqkb", "@33", "asa") // <nil> #1534
        iter13.SeekPrefixGE("zxfrhisycjj@18") // [false] pebble: SeekPrefixGE supplied with key outside of upper bound #1609
        iter18.SeekLT("zxfrhisycjj@5", "") // [true,"tbrm@27","kskrdjghregahqptg",<no range>] <nil> #1655
        --- FAIL: TestMeta/execution/standard-013 (0.21s)
```
<details><summary>Help</summary>
<p>
To reproduce, try:

```bash
go test -tags 'invariants' -exec 'stress -p 1' -timeout 0 -test.v -run TestMeta$ ./internal/metamorphic -seed 1701580677296367537 -ops "uniform:5000-10000"
```
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestMeta.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

github.com/cockroachdb/pebble/internal/metamorphic.TestMeta [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_NightlyPebbleMetamorphicRace/12952561?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_NightlyPebbleMetamorphicRace/12952561?buildTab=artifacts#meta) on refs/heads/master @ [51fca96df08f](https://github.com/cockroachdb/pebble/commits/51fca96df08f):


```
        snap55.Get("raopiz@63") // [""] pebble: not found #5782
        db1.Set("egcrifbw@155", "oqsishbdqavddx") // <nil> #5783
        snap54.Get("egcrifbw@95") // [""] pebble: not found #5784
        batch56 = db1.NewIndexedBatch() #5785
        iter69 = snap55.NewIter("dogogaobvdl", "ecgcfhpkbk", 2 /* key types */, 35, 87, false /* use L6 filters */, "@96" /* masking suffix */) // <nil> #5786
        db1.Set("cedligihpv@133", "") // <nil> #5787
        iter69.Last() // [true,"durzh@71","iiajwzukkyw",["durzh","ecgcfhpkbk")=>{"@138"="x","@106"="c"}*] <nil> #5788
        db1.Ingest(batch56) // <nil> #5789
        iter69.SeekPrefixGE("cedligihpv@120") // [false] pebble: SeekPrefixGE supplied with key outside of lower bound #5790
        iter69.First() // [true,"dogogaobvdl",<no point>,["dogogaobvdl","durzh")=>{"@138"="x","@106"="c","@104"="bvlswakweyjjwpxaun"}*] <nil> #5791
        snap54.Get("egcrifbw@66") // [""] pebble: not found #5797
        iter69.SeekLT("lgybffzpleg@33", "") // [true,"durzh@71","iiajwzukkyw",["durzh","ecgcfhpkbk")=>{"@138"="x","@106"="c"}*] <nil> #5792
        iter69.SeekLT("kbavvkov@74", "") // [true,"durzh@71","iiajwzukkyw",["durzh","ecgcfhpkbk")=>{"@138"="x","@106"="c"}] <nil> #5793
        snap54.Get("egcrifbw@145") // ["zaparlfdgv"] <nil> #5800
        iter69.First() // [true,"dogogaobvdl",<no point>,["dogogaobvdl","durzh")=>{"@138"="x","@106"="c","@104"="bvlswakweyjjwpxaun"}*] <nil> #5794
        iter69.SeekLT("odlpidwwts@142", "") // [true,"durzh@71","iiajwzukkyw",["durzh","ecgcfhpkbk")=>{"@138"="x","@106"="c"}*] <nil> #5795
        iter69.InternalNext() // switching from reverse to forward via internalNext is prohibited #5796
        iter69.Next("") // [false] switching from reverse to forward via internalNext is prohibited #5798
        iter69.SeekLT("fdswnzg@24", "astzpgtppm@52") // [valid,"durzh@71","iiajwzukkyw",["durzh","ecgcfhpkbk")=>{"@138"="x","@106"="c"}*] <nil> #5799
        iter69.First() // [true,"dogogaobvdl",<no point>,["dogogaobvdl","durzh")=>{"@138"="x","@106"="c","@104"="bvlswakweyjjwpxaun"}*] <nil> #5801
        iter69.Prev("") // [false] <nil> #5802
        iter69.Close() // <nil> #5803
        db1.Set("kfcf@158", "oulnfxirhriyt") // <nil> #5804
        db1.Set("rcduzrcz@156", "") // <nil> #5805
        iter70 = db1.NewIter("", "", 2 /* key types */, 0, 0, true /* use L6 filters */, "" /* masking suffix */) // <nil> #5806
        db1.RatchetFormatMajorVersion(010) // <nil> #5809
        db1.Set("rcduzrcz@137", "rezghbtokdemiwjybai") // <nil> #5817
        iter70.SeekPrefixGE("vlormweyhr@38") // [false] <nil> #5807
        db1.Get("egcrifbw@102") // [""] pebble: not found #5824
        db1.Set("fdswnzg@158", "dcjgsjgw") // <nil> #5826
        db1.Set("cedligihpv@45", "nexerbalhksqqajo") // <nil> #5837
        iter70.SeekLT("vlormweyhr@137", "kbavvkov@11") // [valid,"vlormweyhr@158","gscuqypfsffoatlvyd",<no range>] <nil> #5808
        iter71 = snap55.NewIter("fdswnzg", "jnvebjgigt", 1 /* key types */, 0, 0, false /* use L6 filters */, "" /* masking suffix */) // <nil> #5840
        iter70.Next("") // [true,"vlormweyhr@137","lcyezypprbapgmxbf",<no range>] <nil> #5810
        iter71.Prev("") // [true,"fdswnzg",<no point>,["fdswnzg","jnvebjgigt")=>{"@106"="c"}*] <nil> #5841
        iter71.Last() // [true,"fdswnzg",<no point>,["fdswnzg","jnvebjgigt")=>{"@106"="c"}] <nil> #5848
        iter71.Prev("") // [false] <nil> #5851
        iter70.Prev("") // [true,"vlormweyhr@158","gscuqypfsffoatlvyd",<no range>] <nil> #5811
        iter71.SeekPrefixGE("vlormweyhr@96") // [false] pebble: SeekPrefixGE supplied with key outside of upper bound #5858
        iter70.InternalNext() // switching from reverse to forward via internalNext is prohibited #5812
        iter70.Prev("") // [false] switching from reverse to forward via internalNext is prohibited #5813
        iter70.SeekPrefixGE("qsjmmvmjg@55") // [true,"qsjmmvmjg@55","wkqepcxppvpahanfhg",["qsjmmvmjg","qsjmmvmjg\x00")=>{"@149"="xyvfcfqfsxkxvlqochv"}*] <nil> #5814
        iter70.SeekLT("kbavvkov@57", "") // [true,"kbavvkov@65","",["egcrifbw","pjqxhpdr")=>{"@149"="xyvfcfqfsxkxvlqochv","@148"="zhmuw"}*] <nil> #5815
        iter70.SeekPrefixGE("vlormweyhr@158") // [true,"vlormweyhr@158","gscuqypfsffoatlvyd",<no range>*] <nil> #5816
        iter70.SeekGE("astzpgtppm@51", "") // [true,"astzpgtppm@51","",["astzpgtppm@63","avxq")=>{"@147"="qvgaebiy","@129"="znvvn","@96"="egfzmrrctwsrtvkpom"}*] <nil> #5818
        iter70.SeekLT("odlpidwwts@116", "") // [true,"odlpidwwts@135","gvuxdc",["egcrifbw","pjqxhpdr")=>{"@149"="xyvfcfqfsxkxvlqochv","@148"="zhmuw"}*] <nil> #5819
        iter70.SeekGE("fdswnzg@117", "") // [true,"fdswnzg@117",<no point>,["egcrifbw","pjqxhpdr")=>{"@149"="xyvfcfqfsxkxvlqochv","@148"="zhmuw"}] <nil> #5820
        iter70.SeekLT("kbavvkov@123", "") // [true,"kbavvkov@136","cwffkeyevzdwolwejq",["egcrifbw","pjqxhpdr")=>{"@149"="xyvfcfqfsxkxvlqochv","@148"="zhmuw"}] <nil> #5821
        iter70.First() // [true,"astzpgtppm",<no point>,["astzpgtppm","astzpgtppm@63")=>{"@147"="qvgaebiy","@129"="znvvn","@123"="yuvleamyfvnu","@116"="f","@115"="yirixgmjalr","@112"="be","@106"="bdchwwpojglaq","@96"="egfzmrrctwsrtvkpom"}*] <nil> #5822
        --- FAIL: TestMeta/execution/standard-018 (2.62s)
```
<details><summary>Help</summary>
<p>
To reproduce, try:

```bash
go test -tags 'invariants' -exec 'stress -p 1' -timeout 0 -test.v -run TestMeta$ ./internal/metamorphic -seed 1701589278340187516 -ops "uniform:5000-10000"
```
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestMeta.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

github.com/cockroachdb/pebble/internal/metamorphic.TestMeta [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_Pebble_Metamorphic/12979150?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_Pebble_Metamorphic/12979150?buildTab=artifacts#meta) on refs/heads/master @ [785907bc7142](https://github.com/cockroachdb/pebble/commits/785907bc7142):


```
        db1.Set("ossapi@9", "uwn") // <nil> #1951
        db1.Get("rvrjaanphap@17") // ["sfhopkbukf"] <nil> #1953
        db1.SingleDelete("abrmrhxxiv@71", false /* maybeReplaceDelete */) // <nil> #1954
        iter17.SeekGE("qljymwe@24", "") // [true,"qljymwe@15","cjyuxsopaafx",<no range>] <nil> #1956
        iter17.Next("") // [true,"qqsmh@61","r",<no range>] <nil> #1958
        db1.Set("gmqiyigv@12", "zfwcvyo") // <nil> #1959
        snap16.Close() // <nil> #1960
        db1.Set("ucqsqm@69", "gkaa") // <nil> #1963
        iter17.Close() // <nil> #1967
        iter20.Close() // <nil> #1968
        iter19.Close() // <nil> #1969
        db1.Restart() #1970
        db1.Set("qqsmh@64", "jnjpifc") // <nil> #1971
        db1.Set("dnjcdiviqlu@54", "zdmzufcytxcqgq") // <nil> #1972
        db1.Get("rvrjaanphap@22") // [""] pebble: not found #1973
        db1.Get("fdcxzzupt@71") // [""] pebble: not found #1974
        db1.Get("gmqiyigv@42") // [""] pebble: not found #1975
        db1.Set("qljymwe@68", "k") // <nil> #1976
        db1.Get("gmqiyigv@75") // ["cna"] <nil> #1977
        db1.Set("slamwnp@82", "lzvibewxwjzxk") // <nil> #1978
        db1.Get("ossapi@11") // [""] pebble: not found #1979
        db1.RangeKeyDelete("qqsmh", "slamwnp") // <nil> #1980
        db1.Get("ujmqhjyvvu@41") // ["wbewtozpwnaqll"] <nil> #1981
        iter22 = db1.NewIter("", "", 2 /* key types */, 0, 0, false /* use L6 filters */, "" /* masking suffix */) // <nil> #1982
        db1.Set("rvrjaanphap@14", "fu") // <nil> #1986
        db1.Delete("rvrjaanphap@81") // <nil> #1988
        db1.Get("rvrjaanphap@23") // [""] pebble: not found #1994
        db1.Set("ujgkj@41", "ewykxwojsogwtsolmbc") // <nil> #1997
        db1.Get("gmqiyigv@15") // [""] pebble: not found #1998
        db1.Get("abrmrhxxiv@37") // ["mygvkef"] <nil> #1999
        iter23 = db1.NewIter("", "rvrjaanphap@38", 1 /* key types */, 0, 0, false /* use L6 filters */, "" /* masking suffix */) // <nil> #2012
        db1.Get("rvrjaanphap@38") // [""] pebble: not found #2014
        iter23.First() // [true,"gmqiyigv",<no point>,["gmqiyigv","ossapi")=>{"@68"="bjwnbvlg"}*] <nil> #2015
        iter23.Close() // <nil> #2016
        // INFO: [JOB 44] WAL created 000031
        // INFO: [JOB 45] flushing 1 memtable (1.7KB) to L0
        // INFO: [JOB 45] flushing: sstable created 000032
        // INFO: [JOB 45] flushed 1 memtable (1.7KB) to L0 [000032] (2.0KB), in 0.0s (0.0s total), output rate 6.4MB/s
        db1.Flush() // <nil> #2021
        db1.Get("slamwnp@29") // ["fvryky"] <nil> #2025
        db1.Set("ujgkj@61", "ftx") // <nil> #2026
        db1.Set("qljymwe@25", "ouxlfetkznwwv") // <nil> #2031
        db1.Set("aozaovv@62", "tzrdfmvlfhjj") // <nil> #2033
        batch14 = db1.NewBatch() #2034
        batch14.RangeKeyUnset("dnjcdiviqlu", "quafr", "@80") // <nil> #2035
        // INFO: [JOB 46] ingesting: sstable created 000033
        iter22.SeekLT("ossapi@11", "") // [true,"ossapi@25","p",["ossapi","qljymwe")=>{"@68"="bjwnbvlg","@1"="bhlfgswukd"}*] <nil> #1983
        // INFO: [JOB 45] WAL deleted 000029
        iter22.SeekGE("qljymwe@48", "qljymwe@44") // [valid,"qljymwe@48",<no point>,["qljymwe","qljymwe@8")=>{"@83"="ndluyp","@68"="bjwnbvlg","@1"="bhlfgswukd"}*] <nil> #1984
        --- FAIL: TestMeta/execution/standard-025 (0.23s)
```
<details><summary>Help</summary>
<p>
To reproduce, try:

```bash
go test -tags 'invariants' -exec 'stress -p 1' -timeout 0 -test.v -run TestMeta$ ./internal/metamorphic -seed 1701754146074353052 -ops "uniform:5000-10000"
```
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestMeta.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

github.com/cockroachdb/pebble/internal/metamorphic.TestMeta [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_NightlyPebbleMetamorphicRace/12979151?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_NightlyPebbleMetamorphicRace/12979151?buildTab=artifacts#meta) on refs/heads/master @ [785907bc7142](https://github.com/cockroachdb/pebble/commits/785907bc7142):


```
        iter24.SetBounds("hklesrxg@6", "uttlbeic@39") // <nil> #2313
        iter24.SeekGE("bpltj@30", "hftbbllo@83") // [invalid] <nil> #2314
        db1.Set("wpzdta@102", "sxzgzeuxqltpepps") // <nil> #2307
        db1.Delete("wpzdta@94") // <nil> #2312
        db1.Set("bpltj@60", "yztjihnxglrnh") // <nil> #2316
        iter27.Close() // <nil> #2319
        snap24.Close() // <nil> #2320
        snap25.Get("icfgasoqxwf@95") // ["e"] <nil> #2329
        db1.Set("ljlcnk@39", "izkgd") // <nil> #2340
        db1.Set("uttlbeic@1", "d") // <nil> #2345
        snap25.Get("ljlcnk@64") // ["cbiihrjsragqftrqi"] <nil> #2352
        db1.Get("hklesrxg@42") // [""] pebble: not found #2357
        iter29 = db1.NewIter("", "", 2 /* key types */, 0, 0, false /* use L6 filters */, "" /* masking suffix */) // <nil> #2367
        db1.Set("uttlbeic@40", "wujsbibrnown") // <nil> #2368
        iter29.SeekPrefixGE("uttlbeic@102") // [true,"uttlbeic@102",<no point>,["uttlbeic","uttlbeic@24")=>{"@78"="wnwyqonchhplxkfcqfg","@64"="rzedat"}*] <nil> #2369
        db1.Set("ljlcnk@29", "rcbwxwnduezh") // <nil> #2370
        db1.Set("ljlcnk@49", "guws") // <nil> #2371
        iter29.First() // [true,"bpltj@105","zkgmrwxgyhllvv",<no range>*] <nil> #2373
        db1.Get("hklesrxg@34") // [""] pebble: not found #2375
        db1.RangeKeyUnset("pbxlaiz", "tbtsyz", "@99") // <nil> #2377
        iter29.SeekLT("uttlbeic@28", "") // [true,"uttlbeic@44","ermjdpoehabjhkc",["uttlbeic","uttlbeic@24")=>{"@78"="wnwyqonchhplxkfcqfg","@64"="rzedat"}*] <nil> #2379
        iter24.SeekLT("uttlbeic@46", "") // [true,"uttlbeic@100","dwbczmxpq",["uttlbeic","uttlbeic@39")=>{"@78"="wnwyqonchhplxkfcqfg","@64"="rzedat"}*] <nil> #2321
        iter24.Next("") // [false] <nil> #2325
        iter24.SeekLT("hklesrxg@52", "") // [false] <nil> #2328
        iter24.Last() // [true,"uttlbeic@100","dwbczmxpq",["uttlbeic","uttlbeic@39")=>{"@78"="wnwyqonchhplxkfcqfg","@64"="rzedat"}*] <nil> #2332
        iter24.SetBounds("", "hklesrxg@86") // <nil> #2333
        iter24.First() // [true,"bpltj",<no point>,["bpltj","gzpqxe")=>{"@102"="k","@64"="rzedat"}*] <nil> #2334
        iter24.InternalNext() // <nil> #2336
        iter24.Prev("") // [false] <nil> #2337
        iter24.SetBounds("uttlbeic@10", "wpzdta@71") // <nil> #2338
        iter24.Next("") // [true,"uttlbeic@10",<no point>,["uttlbeic@10","wpzdta")=>{"@64"="rzedat"}*] <nil> #2341
        iter24.Last() // [true,"uttlbeic@4","ctiftgkbsgkqftaebiw",["uttlbeic@10","wpzdta")=>{"@64"="rzedat"}] <nil> #2342
        iter24.Next("") // [false] <nil> #2343
        iter24.Last() // [true,"uttlbeic@4","ctiftgkbsgkqftaebiw",["uttlbeic@10","wpzdta")=>{"@64"="rzedat"}*] <nil> #2344
        iter24.InternalNext() // switching from reverse to forward via internalNext is prohibited #2346
        iter24.SeekPrefixGE("wpzdta@46") // [false] <nil> #2347
        iter24.SeekGE("wpzdta@19", "") // [false] <nil> #2349
        iter24.Next("") // [false] <nil> #2350
        iter24.Last() // [true,"uttlbeic@4","ctiftgkbsgkqftaebiw",["uttlbeic@10","wpzdta")=>{"@64"="rzedat"}*] <nil> #2358
        iter24.SeekGE("gzpqxe@89", "uttlbeic@79") // [invalid] <nil> #2360
        iter24.Next("") // [true,"uttlbeic@10",<no point>,["uttlbeic@10","wpzdta")=>{"@64"="rzedat"}*] <nil> #2362
        iter24.Next("") // [true,"uttlbeic@4","ctiftgkbsgkqftaebiw",["uttlbeic@10","wpzdta")=>{"@64"="rzedat"}] <nil> #2363
        iter24.Prev("ljlcnk@52") // [valid,"uttlbeic@10",<no point>,["uttlbeic@10","wpzdta")=>{"@64"="rzedat"}] <nil> #2364
        iter24.Next("") // [true,"uttlbeic@4","ctiftgkbsgkqftaebiw",["uttlbeic@10","wpzdta")=>{"@64"="rzedat"}] <nil> #2365
        iter24.InternalNext() // <nil> #2372
        iter24.Next("wpzdta@47") // [invalid] <nil> #2376
        iter24.SeekLT("bpltj@68", "") // [false] <nil> #2378
        iter24.First() // [true,"uttlbeic@10",<no point>,["uttlbeic@10","wpzdta")=>{"@64"="rzedat"}*] <nil> #2383
        iter24.Last() // [true,"uttlbeic@4","ctiftgkbsgkqftaebiw",["uttlbeic@10","wpzdta")=>{"@64"="rzedat"}] <nil> #2387
        --- FAIL: TestMeta/execution/random-017 (1.66s)
```
<details><summary>Help</summary>
<p>
To reproduce, try:

```bash
go test -tags 'invariants' -exec 'stress -p 1' -timeout 0 -test.v -run TestMeta$ ./internal/metamorphic -seed 1701754669881623698 -ops "uniform:5000-10000"
```
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestMeta.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

--- a/batch.go
+++ b/batch.go
@@ -20,6 +20,7 @@ import (
 	"github.com/cockroachdb/pebble/internal/base"
 	"github.com/cockroachdb/pebble/internal/batchskl"
 	"github.com/cockroachdb/pebble/internal/humanize"
+	"github.com/cockroachdb/pebble/internal/invariants"
 	"github.com/cockroachdb/pebble/internal/keyspan"
 	"github.com/cockroachdb/pebble/internal/private"
 	"github.com/cockroachdb/pebble/internal/rangedel"
@@ -931,6 +932,15 @@ func (b *Batch) DeleteRangeDeferred(startLen, endLen int) *DeferredBatchOp {
 //
 // It is safe to modify the contents of the arguments after RangeKeySet returns.
 func (b *Batch) RangeKeySet(start, end, suffix, value []byte, _ *WriteOptions) error {
+	if invariants.Enabled && b.db != nil && b.db.opts.Comparer.Split != nil {
+		// RangeKeySet is only supported on prefix keys.
+		if b.db.opts.Comparer.Split(start) != len(start) {
+			panic("RangeKeySet called with suffixed start key")
+		}
+		if b.db.opts.Comparer.Split(end) != len(end) {
+			panic("RangeKeySet called with suffixed end key")
+		}
+	}
 	suffixValues := [1]rangekey.SuffixValue{{Suffix: suffix, Value: value}}
 	internalValueLen := rangekey.EncodedSetValueLen(end, suffixValues[:])
 
@@ -981,6 +991,15 @@ func (b *Batch) incrementRangeKeysCount() {
 // It is safe to modify the contents of the arguments after RangeKeyUnset
 // returns.
 func (b *Batch) RangeKeyUnset(start, end, suffix []byte, _ *WriteOptions) error {
+	if invariants.Enabled && b.db != nil && b.db.opts.Comparer.Split != nil {
+		// RangeKeyUnset is only supported on prefix keys.
+		if b.db.opts.Comparer.Split(start) != len(start) {
+			panic("RangeKeyUnset called with suffixed start key")
+		}
+		if b.db.opts.Comparer.Split(end) != len(end) {
+			panic("RangeKeyUnset called with suffixed end key")
+		}
+	}
 	suffixes := [1][]byte{suffix}
 	internalValueLen := rangekey.EncodedUnsetValueLen(end, suffixes[:])
 
@@ -1014,6 +1033,15 @@ func (b *Batch) rangeKeyUnsetDeferred(startLen, internalValueLen int) *DeferredB
 // It is safe to modify the contents of the arguments after RangeKeyDelete
 // returns.
 func (b *Batch) RangeKeyDelete(start, end []byte, _ *WriteOptions) error {
+	if invariants.Enabled && b.db != nil && b.db.opts.Comparer.Split != nil {
+		// RangeKeyDelete is only supported on prefix keys.
+		if b.db.opts.Comparer.Split(start) != len(start) {
+			panic("RangeKeyDelete called with suffixed start key")
+		}
+		if b.db.opts.Comparer.Split(end) != len(end) {
+			panic("RangeKeyDelete called with suffixed end key")
+		}
+	}
 	deferredOp := b.RangeKeyDeleteDeferred(len(start), len(end))
 	copy(deferredOp.Key, start)
 	copy(deferredOp.Value, end)

--- a/data_test.go
+++ b/data_test.go
@@ -578,7 +578,7 @@ func runBuildRemoteCmd(td *datadriven.TestData, d *DB, storage remote.Storage) e
 }
 
 func runBuildCmd(td *datadriven.TestData, d *DB, fs vfs.FS) error {
-	b := d.NewIndexedBatch()
+	b := newIndexedBatch(nil, d.opts.Comparer)
 	if err := runBatchDefineCmd(td, b); err != nil {
 		return err
 	}

--- a/db.go
+++ b/db.go
@@ -839,7 +839,6 @@ func (d *DB) applyInternal(batch *Batch, opts *WriteOptions, noSyncWait bool) er
 		if d.split == nil {
 			return errNoSplit
 		}
-		// TODO(jackson): Assert that all range key operands are suffixless.
 	}
 	batch.committing = true
 

--- a/ingest.go
+++ b/ingest.go
@@ -1160,6 +1160,15 @@ func (d *DB) IngestAndExcise(
 	if d.opts.ReadOnly {
 		return IngestOperationStats{}, ErrReadOnly
 	}
+	if invariants.Enabled && d.opts.Comparer.Split != nil {
+		// Excise is only supported on prefix keys.
+		if d.opts.Comparer.Split(exciseSpan.Start) != len(exciseSpan.Start) {
+			panic("IngestAndExcise called with suffixed start key")
+		}
+		if d.opts.Comparer.Split(exciseSpan.End) != len(exciseSpan.End) {
+			panic("IngestAndExcise called with suffixed end key")
+		}
+	}
 	return d.ingest(paths, ingestTargetLevel, shared, exciseSpan, nil /* external */)
 }
 

--- a/metamorphic/generator.go
+++ b/metamorphic/generator.go
@@ -1172,15 +1172,7 @@ func (g *generator) replicate() {
 		dest = g.dbs.rand(g.rng)
 	}
 
-	var startKey, endKey []byte
-	startKey = g.randKeyToRead(0.001) // 0.1% new keys
-	endKey = g.randKeyToRead(0.001)   // 0.1% new keys
-	for g.equal(startKey, endKey) {
-		endKey = g.randKeyToRead(0.01) // 1% new keys
-	}
-	if g.cmp(startKey, endKey) > 0 {
-		startKey, endKey = endKey, startKey
-	}
+	startKey, endKey := g.prefixKeyRange()
 	g.add(&replicateOp{
 		source: source,
 		dest:   dest,
@@ -1451,14 +1443,7 @@ func (g *generator) writerIngestAndExcise() {
 	batchID := g.liveBatches.rand(g.rng)
 	g.removeBatchFromGenerator(batchID)
 
-	start := g.randKeyToWrite(0.001)
-	end := g.randKeyToWrite(0.001)
-	for g.equal(start, end) {
-		end = g.randKeyToWrite(0.001)
-	}
-	if g.cmp(start, end) > 0 {
-		start, end = end, start
-	}
+	start, end := g.prefixKeyRange()
 	derivedDBID := g.objDB[batchID]
 
 	g.add(&ingestAndExciseOp{

--- a/metamorphic/key_manager.go
+++ b/metamorphic/key_manager.go
@@ -406,13 +406,7 @@ func (k *keyManager) update(o op) {
 			k.mergeKeysInto(batchID, s.dbID)
 		}
 		k.checkForDelOrSingleDelTransitionInDB(s.dbID)
-	case *ingestAndExciseOp:
-		// Merge all keys in the batch with the keys in the DB.
-		k.mergeKeysInto(s.batchID, s.dbID)
-		k.checkForDelOrSingleDelTransitionInDB(s.dbID)
-	case *replicateOp:
-		k.mergeKeysInto(s.source, s.dest)
-		k.checkForDelOrSingleDelTransitionInDB(s.dest)
+	// TODO(bilal): Handle replicateOp and ingestAndExciseOp here.
 	case *applyOp:
 		// Merge the keys from this writer into the parent writer.
 		k.mergeKeysInto(s.batchID, s.writerID)

--- a/metamorphic/key_manager.go
+++ b/metamorphic/key_manager.go
@@ -410,6 +410,9 @@ func (k *keyManager) update(o op) {
 		// Merge all keys in the batch with the keys in the DB.
 		k.mergeKeysInto(s.batchID, s.dbID)
 		k.checkForDelOrSingleDelTransitionInDB(s.dbID)
+	case *replicateOp:
+		k.mergeKeysInto(s.source, s.dest)
+		k.checkForDelOrSingleDelTransitionInDB(s.dest)
 	case *applyOp:
 		// Merge the keys from this writer into the parent writer.
 		k.mergeKeysInto(s.batchID, s.writerID)
@@ -522,7 +525,6 @@ func opWrittenKeys(untypedOp op) [][]byte {
 	case *getOp:
 	case *ingestOp:
 	case *ingestAndExciseOp:
-		return [][]byte{t.exciseStart, t.exciseEnd}
 	case *initOp:
 	case *iterFirstOp:
 	case *iterLastOp:
@@ -550,7 +552,6 @@ func opWrittenKeys(untypedOp op) [][]byte {
 	case *singleDeleteOp:
 		return [][]byte{t.key}
 	case *replicateOp:
-		return [][]byte{t.start, t.end}
 	}
 	return nil
 }

--- a/metamorphic/ops.go
+++ b/metamorphic/ops.go
@@ -821,7 +821,7 @@ func (o *ingestAndExciseOp) run(t *test, h historyRecorder) {
 	err = firstError(err, err2)
 	err = firstError(err, b.Close())
 
-	if writerMeta.Properties.NumEntries == 0 {
+	if writerMeta.Properties.NumEntries == 0 && writerMeta.Properties.NumRangeKeys() == 0 {
 		// No-op.
 		h.Recordf("%s // %v", o, err)
 		return
@@ -1585,7 +1585,7 @@ func (r *replicateOp) runSharedReplicate(
 		h.Recordf("%s // %v", r, err)
 		return
 	}
-	if len(sharedSSTs) == 0 && meta.Properties.NumEntries == 0 {
+	if len(sharedSSTs) == 0 && meta.Properties.NumEntries == 0 && meta.Properties.NumRangeKeys() == 0 {
 		// IngestAndExcise below will be a no-op. We should do a
 		// DeleteRange+RangeKeyDel to mimic the behaviour of the non-shared-replicate
 		// case.

